src/envoy/http/jwt_auth/jwt_authenticator.cc

```diff
@@ -196,10 +196,15 @@ void JwtAuthenticator::VerifyKey(const PubkeyCacheItem& issuer_item) {
   }
 
   headers_->addReferenceKey(kJwtPayloadKey, jwt_->PayloadStrBase64Url());
-  const LowerCaseString key(issuer_item.jwt_config().forward_payload_header());
-  if (!key.get().empty() && !jwt_->PayloadStrBase64Url().empty() &&
-      !(key == kJwtPayloadKey)) {
-    headers_->addCopy(key, jwt_->PayloadStrBase64Url());
+
+  if (issuer_item.jwt_config().forward_payload_header().empty() == false) {
+    const LowerCaseString key(
+        issuer_item.jwt_config().forward_payload_header());
+    // No need to check jwt_->PayloadStrBaseUrl(): a valid JWT will not have an
+    // empty Payload
+    if (!(key == kJwtPayloadKey)) {
```

Comment:
another readability issue.  I don't like  if ( !(A == B))    

if (A != B),

if LowerCaseString doesn't allow you to do that,   use std::string.

Reply:
Can you remind me what is kJwtPayloadKey again? I thought we remove this constant?

Reply:
Changed to if (key.get() != kJwtPayloadKey.get()). kJwtPayloadKey is the original output from jwt_auth. It is preserved for backward compatibility (e.g., not breaking legacy backends using it).